guides/source/action_controller_advanced_topics.md

```diff
@@ -0,0 +1,459 @@
+```
+
+Matching URLs will be replaced with '[FILTERED]'. However, if you only wish to filter the parameters, not the whole URLs,
+please take a look at [Parameters Filtering](#parameters-filtering).
+
+Force HTTPS Protocol
+--------------------
+
+If you'd like to ensure that communication to your controller is only possible
+via HTTPS, you should do so by enabling the [`ActionDispatch::SSL`][] middleware via
+[`config.force_ssl`][] in your environment configuration.
+
+[`config.force_ssl`]: configuring.html#config-force-ssl
+[`ActionDispatch::SSL`]: https://api.rubyonrails.org/classes/ActionDispatch/SSL.html
+
+Built-in Health Check Endpoint
+------------------------------
+
+Rails also comes with a built-in health check endpoint that is reachable at the `/up` path. This endpoint will return a 200 status code if the app has booted with no exceptions, and a 500 status code otherwise.
+
+In production, many applications are required to report their status upstream, whether it's to an uptime monitor that will page an engineer when things go wrong, or a load balancer or Kubernetes controller used to determine a pod's health. This health check is designed to be a one-size fits all that will work in many situations.
+
+While any newly generated Rails applications will have the health check at `/up`, you can configure the path to be anything you'd like in your "config/routes.rb":
+
+```ruby
+Rails.application.routes.draw do
+  get "healthz" => "rails/health#show", as: :rails_health_check
+end
+```
+
+The health check will now be accessible via the `/healthz` path.
+
+NOTE: This endpoint does not reflect the status of all of your application's dependencies, such as the database or redis cluster. Replace "rails/health#show" with your own controller action if you have application specific needs.
+
+Think carefully about what you want to check as it can lead to situations where your application is being restarted due to a third-party service going bad. Ideally, you should design your application to handle those outages gracefully.
+
+Rescue
+------
+
+Most likely your application is going to contain bugs or otherwise throw an exception that needs to be handled. For example, if the user follows a link to a resource that no longer exists in the database, Active Record will throw the `ActiveRecord::RecordNotFound` exception.
+
+Rails default exception handling displays a "500 Server Error" message for all exceptions. If the request was made locally, a nice traceback and some added information gets displayed, so you can figure out what went wrong and deal with it. If the request was remote Rails will just display a simple "500 Server Error" message to the user, or a "404 Not Found" if there was a routing error, or a record could not be found. Sometimes you might want to customize how these errors are caught and how they're displayed to the user. There are several levels of exception handling available in a Rails application:
```

Comment:
This seems to be the only occurence of `traceback` instead of `backtrace` in the Rails repo.
Also the usage of `locally` and `remote` are a bit confusing here:
```suggestion
Rails default exception handling displays a "500 Server Error" message for all exceptions. If the request was made in development a nice backtrace and some added information gets displayed, so you can figure out what went wrong and deal with it. If the request is made in production, Rails will display a simple "500 Server Error" message to the user, or a "404 Not Found" if there was a routing error, or a record could not be found. Sometimes you might want to customize how these errors are caught and how they're displayed to the user. There are several levels of exception handling available in a Rails application:
```

Also this should probably mention the [config.action_dispatch.show_exceptions](https://edgeguides.rubyonrails.org/configuring.html#config-action-dispatch-show-exceptions)?